### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Sparix**

### CODE REVIEW

**Comment by Sparix** [comment]
image1
image

**Comment by Sparix** [comment]
image2
image3

File: taxi/admin.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from django.contrib import admin
```
Comment (Dimosphen1) [inline taxi/admin.py:1]: Exclude `__pycache__` directory with its content from the pull request for all apps.

File: taxi/admin.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+@admin.register(Driver)
+class DriverAdmin(UserAdmin):
+    list_display = UserAdmin.list_display + ("license_number", )
+    fieldsets = UserAdmin.fieldsets + (("Additional info", {"fields": ("license_number", )}), )
+    add_fieldsets = UserAdmin.add_fieldsets + (("Additional info", {"fields": ("license_number", )}), )
+
+
+admin.site.register(Manufacturer)
```
Comment (Dimosphen1) [inline taxi/admin.py]: Use one approach for registering your models, either functional or with the use of decorators.

File: taxi/models.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+class Driver(AbstractUser):
+    license_number = models.CharField(max_length=255, unique=True)
+
+    def __str__(self) -> str:
+        return f"{self.first_name} {self.last_name}"
+
+
+class Car(models.Model):
+    model = models.CharField(max_length=255)
+    manufacturer = models.ForeignKey(Manufacturer, on_delete=models.CASCADE)
```
Comment (Dimosphen1) [inline taxi/models.py]: Add the `related_name` parameter for fields with `ForeignKey`.

File: taxi_service/settings.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/4.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "django-insecure-0!cj3y7$vkr-l(5uw*91=1_u-##icx0!c^n_57ch@gvj+fkf18"
```
Comment (Dimosphen1) [inline taxi_service/settings.py]: Do not store `SECRET_KEY` in a public repository.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+.idea
+venv
+db.sqlite3
+pyc
+__pycache__
```
Comment (Polyakiv-Andrey) [inline .gitignore]: add empty line in the end of file

File: taxi/__init__.py
Comment (Polyakiv-Andrey) [inline taxi/__init__.py:1]: Delete from git .pyc files

**Review by Polyakiv-Andrey — CHANGES_REQUESTED** [review]

File: taxi/admin.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+@admin.register(Driver)
+class DriverAdmin(UserAdmin):
+    list_display = UserAdmin.list_display + ("license_number", )
+    fieldsets = UserAdmin.fieldsets + (("Additional info", {"fields": ("license_number", )}), )
+    add_fieldsets = UserAdmin.add_fieldsets + (("Additional info", {"fields": ("license_number", )}), )
+
+
+@admin.register(Manufacturer)
+class ManufacturerAdmin(admin.ModelAdmin):
+    list_display = ["name", "country", ]
```
Comment (viktoria-rybenchuk) [inline taxi/admin.py]: add a blank line in the end of the file

File: taxi/models.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    def __str__(self) -> str:
+        return f"{self.first_name} {self.last_name}"
+
+
+class Car(models.Model):
+    model = models.CharField(max_length=255)
+    manufacturer = models.ForeignKey(
+        Manufacturer,
+        on_delete=models.CASCADE,
+        related_name="manufacturer"
```
Comment (viktoria-rybenchuk) [inline taxi/models.py]:
```suggestion
        related_name="cars"

```

Comment (viktoria-rybenchuk) [reply taxi/models.py]: When defining related names in Django models, it's a common convention to use the plural form of the related model's class name.

**Review by viktoria-rybenchuk — CHANGES_REQUESTED** [review]
Rollback all changes in test.py. And remove from your repository files as `__pychache__`

**Pull request closed without merging**